patterns/builder.md

```diff
@@ -2,46 +2,49 @@
 
 ## Description
 
-Construct an object with calls to a builder helper. 
-
+Construct an object with calls to a builder helper.
 
 ## Example
 
 ```rust
-struct Foo {
+#[derive(Debug)]
+pub struct Foo {
     // Lots of complicated fields.
+    bar: String,
 }
 
-struct FooBuilder {
+pub struct Builder {
     // Probably lots of optional fields.
-    ...
+    bar: String,
 }
 
-impl FooBuilder {
-    fn new(...) -> FooBuilder {
+impl Builder {
+    pub fn new() -> Builder {
         // Set the minimally required fields of Foo.
+        Builder {
+            bar: "X".to_string(),
+        }
     }
 
-    fn named(mut self, name: &str) -> FooBuilder {
+    pub fn name(mut self, bar: &str) -> Builder {
         // Set the name on the builder itself, and return the builder by value.
+        self.bar = bar.to_string();
+        self
     }
 
-    // More methods that take `mut self` and return `FooBuilder` setting up
-    // various aspects of a Foo.
-    ...
-
     // If we can get away with not consuming the Builder here, that is an
-    // advantage. It means we can use the builder as a template for constructing
-    // many Foos.
-    fn finish(&self) -> Foo {
-        // Create a Foo from the FooBuilder, applying all settings in FooBuilder to Foo.
+    // advantage. It means we can use the builder as a template for constructing many Foos.
+    pub fn finish(self) -> Foo {
```

Comment:
Some parts seemed inconsistent, some uses `finish` and below uses `build`.

Reply:
Unfortunately I had this overwritten, but now it is `build` again 👷